### PR TITLE
Add aks agic addon to managedCluster (AKS)

### PR DESCRIPTION
**Pull request by JPEasier**
# Description
[ContainerService: ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml)

Application Gateway Ingress Controler (AGIC) is a addon configuration for AKS managed Clusster.
- update managedCluster property in bicep
- add new parameter to bicep
- update readme

# Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

**Comment by MrMCake** [comment]
Adding the pipeline reference 
[ContainerService: ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml)

**Comment by MrMCake** [comment]
@JPEasier, note you have a couple conflicts that need resolving

**Comment by JPEasier** [comment]
Wating for Bug: #1347

File: arm/Microsoft.ContainerService/managedClusters/deploy.bicep [hunk 1]
```diff
@@ -328,10 +334,10 @@ var diagnosticsMetrics = [for metric in diagnosticMetricsToEnable: {
 
 var identityType = systemAssignedIdentity ? 'SystemAssigned' : (!empty(userAssignedIdentities) ? 'UserAssigned' : 'None')
 
-var identity = identityType != 'None' ? {
```
Comment (MrMCake) [inline arm/Microsoft.ContainerService/managedClusters/deploy.bicep:331]: Why did you remove that?

**Comment by MrMCake** [comment]
> Correlates to #1275

@JPEasier you can continue without. As nice as linter is - it has some strange rules. For the time being I suggest to apply linter rules per common sense.

File: arm/Microsoft.ContainerService/managedClusters/deploy.bicep (hunk 1 above)
Comment (JPEasier) [reply arm/Microsoft.ContainerService/managedClusters/deploy.bicep:331]: The `identity` block should not be null or empty. You have to set a value for `type`.

File: arm/Microsoft.ContainerService/managedClusters/deploy.bicep [hunk 2]
```diff
@@ -128,6 +128,12 @@ param agentPools array = []
 @description('Optional. Specifies whether the httpApplicationRouting add-on is enabled or not.')
 param httpApplicationRoutingEnabled bool = false
 
+@description('Optional. Specifies whether the ingressApplicationGateway (AGIC) add-on is enabled or not.')
+param ingressApplicationGatewayEnabled bool = false
+
+@description('Conditional. Specifies the resource ID of conected application gateway. Must be set if `ingressApplicationGatewayEnabled` is set to `true`.')
```
Comment (eriqua) [inline arm/Microsoft.ContainerService/managedClusters/deploy.bicep:134]: For consistency with other Conditional descriptions I'd suggest 
```suggestion
@description('Conditional. Required if `ingressApplicationGatewayEnabled` is set to `true`. Specifies the resource ID of connected application gateway.')
```

File: arm/Microsoft.ContainerService/managedClusters/readme.md [hunk 3]
```diff
@@ -26,6 +26,11 @@ This module deploys Azure Kubernetes Cluster (AKS).
 | `name` | string | Specifies the name of the AKS cluster. |
 | `primaryAgentPoolProfile` | array | Properties of the primary agent pool. |
 
+**Conditional parameters**
+| Parameter Name | Type | Default Value | Description |
+| :-- | :-- | :-- | :-- |
+| `appGatewayResourceId` | string | `''` | Specifies the resource ID of conected application gateway. Must be set if `ingressApplicationGatewayEnabled` is set to `true`. |
```
Comment (eriqua) [inline arm/Microsoft.ContainerService/managedClusters/readme.md:32]: Please regenerate readme after changing description as per the other comment
